internal/controllers/topology/cluster/cluster_controller.go

```diff
@@ -202,9 +204,32 @@ func (r *Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (_ ctrl.Re
 func (r *Reconciler) reconcile(ctx context.Context, s *scope.Scope) (ctrl.Result, error) {
 	var err error
 
+	// Get ClusterClass.
+	clusterClass := &clusterv1.ClusterClass{}
+	key := client.ObjectKey{Name: s.Current.Cluster.Spec.Topology.Class, Namespace: s.Current.Cluster.Namespace}
+	if err := r.Client.Get(ctx, key, clusterClass); err != nil {
+		return ctrl.Result{}, errors.Wrapf(err, "failed to retrieve ClusterClass %s", s.Current.Cluster.Spec.Topology.Class)
+	}
+
+	s.Blueprint.ClusterClass = clusterClass
+	// If the ClusterClass `metadata.Generation` doesn't match the `status.ObservedGeneration` requeue as the ClusterClass
+	// is not up to date.
```

Comment:
Maybe good to add a comment that we don't have to requeue as a change in the ClusterClass to observedGeneration will trigger a reconcile automatically.

Otherwise we probably don't remember next week/month/... and then change it

Reply:
Should we filter and not reconcile if the change is just on observed generation? Seems like each ClusterClass reconcile now results in a new update and reconcile?